lib/trento/users/policy.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Trento.Users.Policy do
+  @moduledoc """
+  Policy for the User resource
+
+  User with the ability all:all or users:all can perform any operation on the users.
+  """
+  @behaviour Bodyguard.Policy
+
+  alias Trento.Users.User
+
+  def authorize(action, %User{} = user, User) when action in [:index, :show],
+    do: has_read_ability?(user)
+
+  def authorize(action, %User{} = user, User) when action in [:update, :delete, :create],
+    do: has_write_ability?(user)
+
+  def authorize(_, _, _), do: false
+
+  defp has_write_ability?(user), do: has_global_ability?(user) or has_users_all_ability?(user)
+  defp has_read_ability?(user), do: has_global_ability?(user) or has_users_all_ability?(user)
+
+  defp has_global_ability?(%User{abilities: abilities}),
```

Comment:
I was having a look to this and I thought that it would be cool to have something like this, that is more declarative and you can extend in the future easier with other abilities:
```
defp has_read_ability?(user), do: user_has_any_abilities?(user, [%Ability{name: "all", resource: "all"}, %Ability{name: "users", resource: "all"}])

def user_has_any_abilitities(%User{abilities: abilities}, expected_abilities), 
    do: Enum.any?(expected_abilities, fn expected_ability -> expected_ability in abilities end)
```

Just my head thinking, feel free to discard hehe

Reply:
I abstracted the ability checking, thx!